src/site/paragraphs/react_widget.drupal.liquid

```diff
@@ -33,6 +33,14 @@
   {% if entity.fieldCtaWidget %}
       <div class="cta-widget" data-widget-type="cta" data-app-id="{{ entity.fieldWidgetType }}"></div>
   {% else %}
+
+    {% assign widgetDelta = 1 %}
+    {% if !reactRoot %}
+      {% assign reactRoot = 'react-widget-' | append: widgetDelta %}
+      {% assign block.entity.reactRoot = reactRoot %}
+      {% assign widgetDelta = widgetDelta | plus: 1 %}
+    {% endif %}
+
```

Comment:
Couldn't you just say:

```
{% if !reactRoot %}
      {% assign reactRoot = 'react-widget-1' %}
{% endif %}
```

Reply:
Doesn't that assume there are no other nested widgets on the page? I was trying to still get the unique id even with multiple nested react widgets, similar to how it works in the page template.

Reply:
But it's inside a for loop in the page template. Wouldn't this code just set `reactRoot` to 'react-widget-1' every time the template's called? 

I think that to make it unique, you'd have to create and increment `widgetDelta` at higher level(s). I'm thinking something like this (pseudocode):

In page template:
``` 
reactRoot = 1
for block in Content {
  if block.type == 'react-widget' || block.type == 'qa-section {
    block.entity.reactRoot = reactRoot
    reactRoot++
  }
}
```
Then in qa-section and qa templates:
```
reactRoot = reactRoot * 10
for block in Content {
  block.entity.reactRoot = reactRoot
  reactRoot++
}
```
This is only guaranteed to be unique if there are less than 10 blocks in each 'Content', but I'm not sure how else to do it in liquid.

Reply:
Updated to implement this. Multiple nested widgets on a page now have unique ids. The only way I was able to get the widget to actually show up was by removing the "static-page-widget" class. Clicked around and confirmed non-nested react widget still work (like on this page: http://localhost:3001/health-care/eligibility/)

Reply:
Sorry, my pseudocode was a mess. In order for it to work properly, you'd need to add the widgetDelta to the block entity as well and use it as a basis for nested widgetDeltas. So the first Q&A Section would be assigned widgetDelta of 1, and its first question would have widgetDelta of 10 (second question 11, etc), and the first question's first answer paragraph would have widgetDelta of 100. The second Q&A Section would get widgetDelta of 2, etc. That way, even if two Q&A sections had widgets in the same place, the ids would still be unique.

Also, note that react widgets can now live directly under Q&A Sections, that is, they can be in `field_questions`, so you'll need to pass them a reactRoot as well.

Reply:
Hi @beeyayjay I tried this but the widgetDelta variable isn't persistent between templates - even when it's assigned to a parameter in the entity. The only fields that remain persistent are the ones that are returned from Drupal as part of the query results for the react widget entity. This PR does give multiple react widgets on different sections of the page unique ids tho so I think I'm confused about the purpose behind the change.